### PR TITLE
[v0.25] Scaffolded many-to-many relationship fixes

### PR DESCRIPTION
**Pull request by cannikin**
When generating a scaffold for a many-to-many relationship, the generator will blow up because it expects the table to have at least one scalar field to create/update in the test file. Many-to-many tables have no scalars (or at least no scalars that aren't foreign keys).

This fix keeps the foreign key columns that are actual scalars (like `userId`) but knows that it can't set this value to a random integer, it needs to be a valid id, which we get from scenario `two`. So the test will look something like:

```javascript
  scenario('creates a tagging', async (scenario) => {
    const result = await createTagging({
      input: {
        postId: scenario.tagging.two.postId,
        tagId: scenario.tagging.two.tagId,
      },
    })

    expect(result.postId).toEqual(scenario.tagging.two.postId)
    expect(result.tagId).toEqual(scenario.tagging.two.tagId)
  })
```

I verified that after generating a scaffold for a one-to-many, many-to-many and no-relations table that the tests all pass!

### CODE REVIEW

**Comment by cannikin** [comment]
We'll want to add docs somewhere to let people know that we can only generate scaffolds for many-to-many tables when you give it a regular `id` column. In the Prisma docs they show this example:

```javascript
model CategoriesOnPosts {
  post        Post     @relation(fields: [postId], references: [id])
  postId      Int
  category    Category @relation(fields: [categoryId], references: [id])
  categoryId  Int
  @@id([postId, categoryId])
}
```

The `@@id` syntax creates an *index* only, which is useless for us. So for scaffolds to work on many-to-many there are two requirements:

1. You use [explicit many-to-many relations] (meaning you define the join table yourself, you don't let Prisma make it for you
2. You add a real `id` to the model, something like:

```javascript
model CategoriesOnPosts {
  id          Int      @id @default(autoincrement())
  post        Post     @relation(fields: [postId], references: [id])
  postId      Int
  category    Category @relation(fields: [categoryId], references: [id])
  categoryId  Int
}
```

## Technical Explanation

We have no easy way to select/update/delete a single record from the DB via a service without an `id`. Unless we go down the massive rabbit hole of letting people somehow find/update/delete records by multiple columns, it's going to be much simpler to just make that a requirement of scaffolds—if you want a scaffold for your table, it needs an `id` column. Done.

**Comment by cannikin** [comment]
Added an error message on failure:

image

**Comment by thedavidprice** [comment]
@cannikin this is working like a dream (well, as much as one can dreams of Scaffolding Prisma Relations that is). Ran through some models for skillz building and got 'em to work. And confirming E2E == ✅

The only thing I'm _not_ seeing in test or code is the nice `@@id is a bad idea` error output above. The error is throwing this:
Screen Shot 2021-02-09 at 6 47 58 PM

Possible you didn't push the **Warning** change to this branch?

**Comment by cannikin** [comment]
Whoops! Just pushed it up!

File: packages/cli/src/commands/generate/sdl/sdl.js [hunk 1]
```diff
@@ -22,6 +24,26 @@ import { files as serviceFiles } from '../service/service'
...
+  const line3 = "you'll need to update your schema definition to include"
+  const line4 = 'an `@id` column. Read more here: '
+  const line5 = chalk.underline.blue(
+    'https://redwoodjs.com/docs/schema-relations'
+  )
+
+  console.error(
+    boxen(line1 + '\n\n' + line2 + '\n' + line3 + '\n' + line4 + '\n' + line5, {
+      padding: 1,
+      margin: 1,
```
Comment (thedavidprice) [inline packages/cli/src/commands/generate/sdl/sdl.js:41]:
```suggestion
      margin: { top: 1, bottom: 3, right: 1, left: 2 },
```
adding bottom margin due to getting cropped

**Pull request merged**